nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/avro/AvroReplacementDescriptor.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.runtime.avro;
+
+import org.nuxeo.common.xmap.annotation.XNode;
+import org.nuxeo.common.xmap.annotation.XObject;
+
+/**
+ * The avro forbidden character replacement descriptor.
+ *
+ * @since 10.2
+ */
+@XObject("replacement")
+public class AvroReplacementDescriptor implements Comparable<AvroReplacementDescriptor> {
+
+    @XNode("@forbidden")
+    private String forbidden;
+
+    @XNode("@replacement")
+    private String replacement;
+
+    @XNode("@priority")
+    private int priority;
+
+    @Override
+    public int compareTo(AvroReplacementDescriptor o) {
+        if (priority != o.priority) {
+            return Integer.compare(priority, o.priority);
+        } else {
+            return forbidden == null ? o.forbidden == null ? 0 : 1 : forbidden.compareTo(o.forbidden);
+        }
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        AvroReplacementDescriptor other = (AvroReplacementDescriptor) obj;
+        if (forbidden == null) {
```

Comment:
Can we test all equality ? with `Objects#equals`, it should be concise

Reply:
Maybe not due to our use of `Set` in the factory ? If so, could you put it as comment in `equals` and `hashCode` ?